cli/tests/pcluster/config/test_param_types.py

```diff
@@ -40,3 +49,56 @@ def test_validate_section_label(self, section_label, should_trigger_error, mocke
             param._validate_section_label()
             for record in caplog.records:
                 assert record.levelname != "ERROR"
+
+
+@pytest.mark.parametrize(
+    "section_dict, expected_value",
+    [
+        ({"volume_size": 100, "ebs_snapshot_id": "snap-1234567890abcdef0"}, 100),
+        ({"ebs_snapshot_id": "snap-1234567890abcdef0"}, 50),
+        ({"volume_size": 100}, 100),
+        ({"volume_size": 0}, 0),
+        ({"volume_size": -5}, -5),
+        ({}, 20),
+    ],
+)
+def test_volume_size_refresh(mocker, boto3_stubber, section_dict, expected_value):
+    mocked_pcluster_config = utils.get_mocked_pcluster_config(mocker)
+    ebs_section = CfnSection(EBS, mocked_pcluster_config, "default")
+    for param_key, param_value in section_dict.items():
+        param = EBS.get("params").get(param_key).get("type", CfnParam)
+        param.value = param_value
+        ebs_section.set_param(param_key, param)
+    mocked_pcluster_config.add_section(ebs_section)
+    config_parser = configparser.ConfigParser()
+    config_parser_dict = {"cluster default": {"ebs_settings": "default"}, "ebs default": section_dict}
+    config_parser.read_dict(config_parser_dict)
+
+    volume_size = VolumeSizeParam(
+        section_key="ebs",
+        section_label="default",
+        param_key="volume_size",
+        param_definition=EBS.get("params").get("volume_size"),
+        pcluster_config=mocked_pcluster_config,
+        owner_section=ebs_section,
+    ).from_file(config_parser)
+
+    if mocked_pcluster_config.get_section("ebs", "default").get_param_value(
+        "volume_size"
+    ) is None and mocked_pcluster_config.get_section("ebs", "default").get_param_value("ebs_snapshot_id"):
```

Comment:
Unlike boto3_stubber, it won't cause errors if you call mocker.patch for a function that never gets called. For that reason, I would get rid of this `if` and always call mocker.patch.

Reply:
Right! Done!